hips/hip-0007.md

```diff
@@ -0,0 +1,69 @@
+---
+hip: 7
+title: "Document and Track subprojects"
+authors: [ "Matt Farina <matt@mattfarina.com>" ]
+created: "2020-11-09"
+type: "process"
+status: "draft"
+---
+
+## Abstract
+
+There are numerous Helm subprojects where each can have zero or more Git
+repositories associated with them. This process provides a centralized method
+to document and track the subprojects, maintainers, and associated repos.
+
+## Motivation
+
+The current design uses a collection of documented and access controls to track
+this information while there are several gaps in the information. The Teams.md
+file attempted to track the different teams (i.e. subprojects, the org
+maintainers, etc). It is unclear what is a subproject, and how the org
+maintainers fit in from that list. The repositories owned by a subproject are
+not documented but are instead tracked via GitHub teams. The owners are
+scattered in various locations that are not consistent. In some cases they
+cannot be consistent (e.g. community management which does not have a GitHub
+repo). This has lead to confusion with the Org Maintainers and others.
+
+## Rationale
+
+Describe why particular design decisions were made.
+
+## Specification
+
+The source of truth will be a YAML file name [`subprojects.yaml`](../subprojects.yaml)
+stored at the root of the Helm community repository. It will have the following
+structure:
+
+```
+subprojects:
+- name: Helm Core
+  ownersLink: https://raw.githubusercontent.com/helm/helm/master/OWNERS
+  repos:
```

Comment:
Should there be an optional mailingList section (or something of the sort) since the Teams.md file mentions some of these mailing lists?  Unless these lists can easily be found elsewhere?

Reply:
I like that idea. I'll fold it in for the next round of changes.

Reply:
done.